rfc/006-genpolicy-platforms.md

```diff
@@ -0,0 +1,111 @@
+# RFC 006: `genpolicy` for diverse platforms
+
+This RFC describes the current situation of `genpolicy` and proposes a strategy for making it work for all supported Contrast platforms.
+
+## Problem Statement
+
+Generated policies are a key building block of Confidential Containers and Contrast.
+However, the only implementation that guarantees basic confidentiality levels is the fork maintained by Microsoft.
+The upstream Kata policy is lacking security fixes that haven't been backported, and yet lacks a mechanism for ensuring image integrity.
+On the other hand, some features that we contributed and would like to use are only available upstream with no ETA of adoption by Microsoft.
+
+We're using the Microsoft fork successfully on AKS-CoCo, but it's tailored to that platform and unlikely to ever be portable.
+Portability will become important for our bare-metal efforts, though.
+Bare-metal cluster will diverge from the standardized AKS environment, requiring flexibility in policy generation.
+
+## Parts of genpolicy
+
+The workload policy mechanism can be divided into four parts:
+
+* The tool that generates reference values from workload YAML and composes the workload policy (`genpolicy`).
+* The settings file for the tool (`genpolicy-settings.json`).
+* The policy rules themselves (`rules.rego`).
+* The policy engine that's queried in the agent.
+
+These components are somewhat independent, and the following subsections describe how they fit together.
+
+### The tool
+
+The idea of the `genpolicy` tool is to take a set of Kubernetes workload descriptors and derive the Kata agent API requests that a legitimate runtime would issue to spawn these workloads.
+It appends these anticipated requests as data to the rules file to produce a complete policy.
+
+Guessing the transformations from Kubernetes to CRI is easy because it's somewhat stable, and the path from Kata runtime to the agent is under our control.
```

Comment:
```suggestion
Guessing the transformations from Kubernetes to CRI is easy because it's somewhat stable, and the path from Kata runtime to the agent is under our control.
```
I don't really get this. What does it mean?

Reply:
Let me try to explain this better.